Let it snow

yubiuser opened the pull request:
**What does this PR aim to accomplish?:**

https://twitter.com/squidfunk/status/1597939243090788352

---
**By submitting this pull request, I confirm the following:**

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against.
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
6. I have checked that another pull request for this purpose does not exist.
7. I have considered, and confirmed that this submission will be valuable to others.
8. I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
9. I give this submission freely, and claim no ownership to its content.

---
- [ ] I have read the above and my PR is ready for review. *Check this box to confirm*

yubiuser commented:
Closing this PR.

While it's a nice winter easter egg, one needs to include the "HoHoHo" meta information on **each** page unless we would use the `meta` plugin  - which is only available to sponsors.

https://squidfunk.github.io/mkdocs-material/reference/#built-in-meta-plugin

DL6ER commented:
I like the idea. When you navigate across the site, the snowing stays. I'm good with adding snowing to the start page and let it snow while users continue to browse the site. If they come here from somewhere else (e.g. directly to the unbound guide), I don't see any issue with no snow falling there.

DL6ER commented:
![Screenshot from 2022-12-04 10-25-47](https://user-images.githubusercontent.com/16748619/205483427-88de563f-18ef-4c23-9d88-2f0a5975879b.png)

yubiuser commented:
I added the code to the `index.md` and rebased.
__

Maybe it's only my old laptop, but I get Chrome using 100% CPU on the Netlify preview when it snows.

DL6ER commented:
I see about 5% CPU increase (some old Xeon E3 on this machine) when viewing the documentation. While this may be a nice feature (get warm and comfortable because of extra CPU exhaust heat) it is also a waste of energy for just some nice effect. Given this, I vote for closing.

PromoFaux commented:
Ah-hah, this reminds me:

![image](https://user-images.githubusercontent.com/1998970/206291834-92c67453-6d49-42c4-a65b-1dc44ce7b63e.png)

![image](https://user-images.githubusercontent.com/1998970/206291865-7c8c600f-8b21-43d9-b1a6-6e797fe50f72.png)

I usually switch them on about a week before Christmas and then turn them off again soon after.

rdwebdesign commented:
> I see about 5% CPU increase (some old Xeon E3 on this machine) when viewing the documentation. While this may be a nice feature (get warm and comfortable because of extra CPU exhaust heat) it is also a waste of energy for just some nice effect. Given this, I vote for closing.

Here, I saw an increase around 8%-10% (i7 processor).
It's a low increase, but it is a waste of energy, specially if you consider that the effect is almost invisible using the light theme.

rdwebdesign commented:
Is there a way to install the snow effect option, but load the page with the option turned off?

The users can click on the button and start the snow, if they like it (but I'm not sure if this is worth it).

![snow_on_off](https://user-images.githubusercontent.com/1385443/206296339-33171c85-0dd0-4b8d-b8be-a6cdd4512422.gif)

yubiuser commented:
Increasing CPU load, even if it's just 5%, for such a little gain should not happen nowadays. Closing until next year. Maybe there will be an updated version which is not that resource intensive.
